#Data Collection/preparation part.

1.The population of each state.

2.Literacy Rate in each state

3.Area of each state

4.Collect any other data that helps with your analysis. There is no limitation for anything.

5.Create a new file and keep the above-collected data.

#source of data  -https://en.wikipedia.org/wiki/List_of_states_and_union_territories_of_India_by_population

# Importing Libraries

In [2]:
import pandas as pd
import numpy as np

# Importing dataset

1.The population of each state.

In [3]:
Population=pd.read_csv("List of states and union territories of India by population 2011 census of india.csv")

Population


sr.no                  State or Union Territory  Population]
0       1               Andaman and Nicobar Islands       380581
1       2                            Andhra Pradesh     49577103
2       3                         Arunachal Pradesh      1383727
3       4                                     Assam     31205576
4       5                                     Bihar    104099452
5       6                                Chandigarh      1055450
6       7                              Chhattisgarh     25545198
7       8  Dadra and Nagar Haveli and Daman and Diu       585764
8       9                                  Delhi UT     16787941
9      10                                       Goa      1458545
10     11                                   Gujarat     60439692
11     12                                   Haryana     25351462
12     13                          Himachal Pradesh      6864602
13     14                         Jammu and Kashmir     12267032
14     15                                 Jharkhand     32988134
15     16                                 Karnataka     61095297
16     17                                    Kerala     33406061
17     18                               Lakshadweep        64473
18     19                            Madhya Pradesh     72626809
19     20                               Maharashtra    112374333
20     21                                Manipur[c]      2570390
21     22                                 Meghalaya      2966889
22     23                                   Mizoram      1097206
23     24                                  Nagaland      1978502
24     25                                    Odisha     41974219
25     26                                Puducherry      1247953
26     27                                    Punjab     27743338
27     28                                 Rajasthan     68548437
28     29                                    Sikkim       610577
29     30                                Tamil Nadu     72147030
30     31                                   Tripura      3673917
31     32                             Uttar Pradesh    199812341
32     33                               Uttarakhand     10086292
33     34                               West Bengal     91276115

We observe that there are total 27 states and 7 union territories in india and highest populated state in india is Uttar Pradesh with population 199812341. as per 2011 census

2.Literacy Rate in each state

#SOURCE - https://www.findeasy.in/indian-states-by-literacy-rate/#State_Wise_Literacy_Rate_in_India

In [6]:
Literacy_rate=pd.read_csv("Literacy Rate in each state.csv")

Literacy_rate

sr.no                  State or Union Territory  Population]  Literates  \
0       1               Andaman and Nicobar Islands       380581     294281   
1       2                            Andhra Pradesh     49577103   50556760   
2       3                         Arunachal Pradesh      1383727     766005   
3       4                                     Assam     31205576   19177977   
4       5                                     Bihar    104099452   52504553   
5       6                                Chandigarh      1055450     805438   
6       7                              Chhattisgarh     25545198   15379922   
7       8  Dadra and Nagar Haveli and Daman and Diu       585764     223230   
8       9                                  Delhi UT     16787941    1342434   
9      10                                       Goa      1458545     188406   
10     11                                   Gujarat     60439692   12737767   
11     12                                   Haryana     25351462    1165487   
12     13                          Himachal Pradesh      6864602   41093358   
13     14                         Jammu and Kashmir     12267032   16598988   
14     15                                 Jharkhand     32988134    5039736   
15     16                                 Karnataka     61095297    7067233   
16     17                                    Kerala     33406061   18328069   
17     18                               Lakshadweep        64473   28135824   
18     19                            Madhya Pradesh     72626809      52553   
19     20                               Maharashtra    112374333   42851169   
20     21                                Manipur[c]      2570390   81554290   
21     22                                 Meghalaya      2966889    1908476   
22     23                                   Mizoram      1097206    1785005   
23     24                                  Nagaland      1978502     848175   
24     25                                    Odisha     41974219   26742595   
25     26                                Puducherry      1247953     957309   
26     27                                    Punjab     27743338   18707137   
27     28                                 Rajasthan     68548437   38275282   
28     29                                    Sikkim       610577     444952   
29     30                                Tamil Nadu     72147030   51837507   
30     31                                   Tripura      3673917    2804783   
31     32                             Uttar Pradesh    199812341  114397555   
32     33                               Uttarakhand     10086292    6880953   
33     34                               West Bengal     91276115   61538281   

    Literacy  
0      86.63  
1      67.02  
2      65.38  
3      72.19  
4      61.80  
5      70.28  
6      86.05  
7      81.24  
8      86.21  
9      88.70  
10     78.03  
11     75.55  
12     82.80  
13     67.16  
14     66.41  
15     75.36  
16     94.00  
17     69.32  
18     82.34  
19     76.94  
20     74.43  
21     91.33  
22     79.55  
23     91.33  
24     72.87  
25     85.85  
26     75.84  
27     66.11  
28     81.42  
29     80.09  
30     87.22  
31     78.82  
32     67.68  
33     76.36

WE observe that out of all states highest literacy rate in Kerala with 94.00%

3.Area of each state Area of each state.csv

#SOURCE -https://en.wikipedia.org/wiki/List_of_states_and_union_territories_of_India_by_area

In [5]:
Area=pd.read_csv("Area of each state.csv")

Area

sr.no                  State or Union Territory  Area (km2)  \
0       1               Andaman and Nicobar Islands        8249   
1       2                            Andhra Pradesh      162975   
2       3                         Arunachal Pradesh       83743   
3       4                                     Assam       78438   
4       5                                     Bihar       94163   
5       6                                Chandigarh         114   
6       7                              Chhattisgarh      135192   
7       8  Dadra and Nagar Haveli and Daman and Diu         603   
8       9                                  Delhi UT       16579   
9      10                                       Goa        1484   
10     11                                   Gujarat        3702   
11     12                                   Haryana      196024   
12     13                          Himachal Pradesh       44212   
13     14                         Jammu and Kashmir       55673   
14     15                                 Jharkhand       42241   
15     16                                 Karnataka       79716   
16     17                                    Kerala      191791   
17     18                               Lakshadweep       59146   
18     19                            Madhya Pradesh          32   
19     20                               Maharashtra      308252   
20     21                                Manipur[c]      307713   
21     22                                 Meghalaya       22327   
22     23                                   Mizoram       22429   
23     24                                  Nagaland       21081   
24     25                                    Odisha      155707   
25     26                                Puducherry         479   
26     27                                    Punjab       50362   
27     28                                 Rajasthan      342239   
28     29                                    Sikkim        7096   
29     30                                Tamil Nadu      130058   
30     31                                   Tripura       10491   
31     32                             Uttar Pradesh      240928   
32     33                               Uttarakhand       53483   
33     34                               West Bengal       88752   

    National share (%)  
0                0.251  
1                4.957  
2                2.548  
3                2.386  
4                2.864  
5                0.003  
6                4.112  
7                0.018  
8                0.505  
9                0.045  
10               0.112  
11               5.962  
12               1.344  
13               1.693  
14               1.285  
15               2.424  
16               5.833  
17               1.800  
18               0.001  
19               9.376  
20               9.359  
21               0.680  
22               0.682  
23               0.642  
24               4.736  
25               0.015  
26               1.531  
27              10.414  
28               0.217  
29               3.956  
30               0.320  
31               7.328  
32               1.626  
33               2.700

In [ ]:
We observe that rajasthan state is at bigest state in area with 342239 km2 

4.Collect any other data that helps with your analysis. There is no limitation for anything.

5.Create a new file and keep the above-collected data.

Sourse-https://en.wikipedia.org/wiki

In [4]:
Population_literacy_area=pd.read_csv("List of states and union territories of India by population,literacy and area.csv")

Population_literacy_area


sr.no                  State or Union Territory  Population]  Literates  \
0       1               Andaman and Nicobar Islands       380581     294281   
1       2                            Andhra Pradesh     49577103   50556760   
2       3                         Arunachal Pradesh      1383727     766005   
3       4                                     Assam     31205576   19177977   
4       5                                     Bihar    104099452   52504553   
5       6                                Chandigarh      1055450     805438   
6       7                              Chhattisgarh     25545198   15379922   
7       8  Dadra and Nagar Haveli and Daman and Diu       585764     223230   
8       9                                  Delhi UT     16787941    1342434   
9      10                                       Goa      1458545     188406   
10     11                                   Gujarat     60439692   12737767   
11     12                                   Haryana     25351462    1165487   
12     13                          Himachal Pradesh      6864602   41093358   
13     14                         Jammu and Kashmir     12267032   16598988   
14     15                                 Jharkhand     32988134    5039736   
15     16                                 Karnataka     61095297    7067233   
16     17                                    Kerala     33406061   18328069   
17     18                               Lakshadweep        64473   28135824   
18     19                            Madhya Pradesh     72626809      52553   
19     20                               Maharashtra    112374333   42851169   
20     21                                Manipur[c]      2570390   81554290   
21     22                                 Meghalaya      2966889    1908476   
22     23                                   Mizoram      1097206    1785005   
23     24                                  Nagaland      1978502     848175   
24     25                                    Odisha     41974219   26742595   
25     26                                Puducherry      1247953     957309   
26     27                                    Punjab     27743338   18707137   
27     28                                 Rajasthan     68548437   38275282   
28     29                                    Sikkim       610577     444952   
29     30                                Tamil Nadu     72147030   51837507   
30     31                                   Tripura      3673917    2804783   
31     32                             Uttar Pradesh    199812341  114397555   
32     33                               Uttarakhand     10086292    6880953   
33     34                               West Bengal     91276115   61538281   

    Literacy  Area (km2)  National share (%)  
0      86.63        8249               0.251  
1      67.02      162975               4.957  
2      65.38       83743               2.548  
3      72.19       78438               2.386  
4      61.80       94163               2.864  
5      70.28         114               0.003  
6      86.05      135192               4.112  
7      81.24         603               0.018  
8      86.21       16579               0.505  
9      88.70        1484               0.045  
10     78.03        3702               0.112  
11     75.55      196024               5.962  
12     82.80       44212               1.344  
13     67.16       55673               1.693  
14     66.41       42241               1.285  
15     75.36       79716               2.424  
16     94.00      191791               5.833  
17     69.32       59146               1.800  
18     82.34          32               0.001  
19     76.94      308252               9.376  
20     74.43      307713               9.359  
21     91.33       22327               0.680  
22     79.55       22429               0.682  
23     91.33       21081               0.642  
24     72.87      155707               4.736  
25     85.85    

#source https://en.wikipedia.org/wiki and India_Crime - Copy.zip

crime type and total crime with population and literacy rate from 2001 to 2012

In [8]:
Crime_rate=pd.read_csv("different type of crime rate statewise with literacy and population from 2001 to 2011.csv")

Crime_rate

sr.no                  State or Union Territory  Population]  Population  \
0       1               Andaman and Nicobar Islands       380581      380581   
1       2                            Andhra Pradesh     49577103    84580777   
2       3                         Arunachal Pradesh      1383727     1383727   
3       4                                     Assam     31205576    31205576   
4       5                                     Bihar    104099452   104099452   
5       6                                Chandigarh      1055450     1055450   
6       7                              Chhattisgarh     25545198    25545198   
7       8  Dadra and Nagar Haveli and Daman and Diu       585764      343709   
8       9                                  Delhi UT     16787941     1978502   
9      10                                       Goa      1458545      243247   
10     11                                   Gujarat     60439692    16787941   
11     12                                   Haryana     25351462     1458545   
12     13                          Himachal Pradesh      6864602    60439692   
13     14                         Jammu and Kashmir     12267032    25351462   
14     15                                 Jharkhand     32988134     6864602   
15     16                                 Karnataka     61095297    12541302   
16     17                                    Kerala     33406061    32988134   
17     18                               Lakshadweep        64473    33406061   
18     19                            Madhya Pradesh     72626809       64473   
19     20                               Maharashtra    112374333    72626809   
20     21                                Manipur[c]      2570390   112374333   
21     22                                 Meghalaya      2966889     2855794   
22     23                                   Mizoram      1097206     2966889   
23     24                                  Nagaland      1978502     1097206   
24     25                                    Odisha     41974219    41974218   
25     26                                Puducherry      1247953     1247953   
26     27                                    Punjab     27743338    27743338   
27     28                                 Rajasthan     68548437    68548437   
28     29                                    Sikkim       610577      610577   
29     30                                Tamil Nadu     72147030    72147030   
30     31                                   Tripura      3673917     3673917   
31     32                             Uttar Pradesh    199812341   199812341   
32     33                               Uttarakhand     10086292    10086292   
33     34                               West Bengal     91276115    91276115   

    Literates Literacy  Area (km2)  National share (%)  Sum of MURDER  \
0      294281   86.63%        8249               0.251            320   
1    50556760   67.02%      162975               4.957          63512   
2      766005   65.38%       83743               2.548           1682   
3    19177977   72.19%       78438               2.386          30864   
4    52504553   61.80%       94163               2.864          82490   
5      805438   86.05%         114               0.003            486   
6    15379922   70.28%      135192               4.112          24120   
7      223230   76.24%         603               0.018            352   
8     1342434   79.55%       16579               0.505          12410   
9      188406   87.10%        1484               0.045            974   
10   12737767   86.21%        3702               0.112          27550   
11    1165487   88.70%      196024               5.962          20942   
12   41093358   78.03%       44212               1.344           2908   
13   16598988   75.55%       55673               1.693          12608   
14    5039736   82.80%       42241               1.285          38120   
15    7067233   67.16%       79716 